File: Projects/C/lab6_pd.c
#include "stdio.h"
#include "stdlib.h"
#include "time.h"
#include "windows.h"

static int random;
static int attempt;
static int input;

void draw()
{
    srand(time(0));
    random = (rand() % 100 + 1);
    printf("\nThe program has chosen a number from 1 to 100, now you can start guessing.");
}

void checking()
{
    printf("\nEnter a number: ");

    if (scanf("%i", &input) == 1)
    {
        if(input < random)
        {
            attempt++;
            printf("\nThe clue number is higher than the entered one. Keep guessing.\nNumber of attempts: %i\n", attempt);
            checking();
        }
        else if(input > random)
        {
            attempt++;
            printf("\nThe clue number is lower than the entered one. Keep guessing.\nNumber of attempts: %i\n", attempt);
            checking();
        }
        else
        {
            attempt++;
            system("cls");
            system("Color A");
            printf("Congratulations! The clue number was %i.\nYou have guessed it within %i attempts.\n\n", random, attempt);
            system("pause");
        }
    }
    else
    {
        system("cls");
        system("Color 4");
        printf("Haha, i've secured it - this is not an integer number.\nYou've ruined the game.\n\n");
        system("pause");
    }
}

int main()
{
    system("Color 4"); //czerwony
    printf("\n    ____       _       __  __      _____\n");
    Sleep(250);
    system("Color E"); //zółty
    printf("U  / ___| UU  / \\  UU |  \\/  | UU | ____| U\n");
    Sleep(250);
    system("Color A"); //zielony
    printf(" \\| |  _ /  \\/ _ \\/  \\| |\\/| |/  \\|  _|  /\n");
    Sleep(250);
    system("Color B"); //cyjan
    printf("  | |_| |   / ___ \\   | |  | |    | |___\n");
    Sleep(250);
    system("Color 9"); //niebieski
    printf("   \\____|  /_/   \\_\\  |_|  |_|    |_____|\n");
    Sleep(250);
    system("Color D"); //różowy
    printf("   _||||_   \\\\   //   _||  ||_   _//   \\\\_\n");
    Sleep(250);
    system("Color F"); //jasny biały
    printf("  (__)(__) (__) (__) (__)  (__) (__)   (__)\n");
    Sleep(250);
    system("Color 7"); //zwykły biały

    draw();
    checking();

    return 0;
}

File: Projects/Python/lab6_pd.py
import random
import time
import os

os.system("cls")  #dodane cls na początek, bo bez tego kolorowanie ascii się czasami bugowało
print("\33[31m    ____       _       __  __      _____")  # czerwony
time.sleep(0.25)
print("\33[33mU  / ___| UU  / \\  UU |  \\/  | UU | ____| U")  # żółty
time.sleep(0.25)
print("\33[92m \\| |  _ /  \\/ _ \\/  \\| |\\/| |/  \\|  _|  /")  # zielony
time.sleep(0.25)
print("\33[96m  | |_| |   / ___ \\   | |  | |    | |___")  # cyjan
time.sleep(0.25)
print("\33[94m   \\____|  /_/   \\_\\  |_|  |_|    |_____|")  # niebieski
time.sleep(0.25)
print("\33[95m   _||||_   \\\\   //   _||  ||_   _//   \\\\_")  # fioletowy
time.sleep(0.25)
print("\33[97m  (__)(__) (__) (__) (__)  (__) (__)   (__)")  # jasny biały
time.sleep(0.25)

random_num = random.randrange(1, 100)
print("\33[0m\nThe program has chosen a number from 1 to 100, now you can start guessing.")

try:
    input_num = int(input("Enter a number: "))
    attempt = 0
    while input_num != random_num:
        if input_num < random_num:
            attempt += 1
            print(f"\nThe clue number is higher than the entered one. Keep guessing.\nNumber of attempts: {attempt}\n")
        elif input_num > random_num:
            attempt += 1
            print(f"\nThe clue number is lower than the entered one. Keep guessing.\nNumber of attempts: {attempt}\n")
        input_num = int(input("Enter a number: "))
    attempt += 1
    os.system("cls")
    print(f"\33[92mCongratulations! The clue number was {random_num}.\nYou have guessed it within {attempt} attempts.\n\n")
    os.system("pause")
except ValueError:
    os.system("cls")
    print("\33[91mHaha, i've secured it - this is not an integer number.\nYou've ruined the game.\n\n")
    os.system("pause")
